src/main/kotlin/com/openlattice/search/SortDefinition.kt

```diff
@@ -0,0 +1,37 @@
+package com.openlattice.search
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.google.common.base.Preconditions
+import com.openlattice.client.serialization.SerializationConstants
+import java.util.*
+
+data class SortDefinition(
+        @JsonProperty(SerializationConstants.TYPE_FIELD) val sortType: SortType = SortType.score,
+        @JsonProperty(SerializationConstants.IS_DESCENDING) val isDescending: Boolean = true,
+        @JsonProperty(SerializationConstants.PROPERTY_TYPE_ID) val propertyTypeId: UUID? = null,
```

Comment:
Why not use Optional for the nullable fields?

Reply:
yeahhh I like being able to use the kotlin default values but we can still do that with optionals / I guess it's more in line with everything else

Reply:
I think we should choose one or the other to standardize on going forward. I'm never really sure which to use but I generally prefer `<TYPE>?` over `Optional<TYPE>`

Reply:
With java always use Optional<T>. 

In Kotlin, if it is not nullable then just use T, possibly with a default value. If it is nullable use T? over Optional<T>.